code/modules/mob/living/living_movement.dm

```diff
@@ -28,9 +28,12 @@
 
 /mob/living/proc/update_turf_movespeed(turf/open/T)
 	if(isopenturf(T))
-		add_or_update_variable_movespeed_modifier(/datum/movespeed_modifier/turf_slowdown, multiplicative_slowdown = T.slowdown)
-	else
+		if(T.slowdown != current_turf_slowdown)
```

Comment:
the direct movespeed modifier setters are surprisingly costly but since 99% of the time theres no movespeed modification from turfs and 99.999% of the time the movespeed modifier of the last turf is the same as the new turf, we can save almost the entire cost of this proc by just only updating it when its different